core/editor/src/document/document_message_handler.rs

```diff
@@ -160,6 +161,13 @@ impl MessageHandler<DocumentMessage, ()> for DocumentMessageHandler {
 					responses.push_back(DocumentOperation::DeleteLayer { path }.into())
 				}
 			}
+			DuplicateSelectedLayers => {
+				// TODO: Replace with drain_filter https://github.com/rust-lang/rust/issues/59618
+				let paths: Vec<Vec<LayerId>> = self.active_document().layer_data.iter().filter_map(|(path, data)| data.selected.then(|| path.clone())).collect();
+				for path in paths {
+					responses.push_back(DocumentOperation::DuplicateLayer { path }.into())
+				}
```

Comment:
You could avoid the collection into a vec to reduce heap allocations

Reply:
We have to collect it to a vec for deleting layers, because we can't iterate through a collection and simultaneously remove things from  it (hence the drain_filter comment). But in this case that is not necessary, as we only iterate over a non mutable reference